### PR TITLE
refactor: separate files from replication.thrift

### PR DESCRIPTION
**Pull request by neverchanje**
This PR separates the replication.thrift into multiple files:

- src/common/backup.thrift
- src/common/bulkload.thrift
- src/common/consensus.thrift
- src/common/duplication.thrift
- src/common/meta_admin.thrift
- src/common/metadata.thrift
- src/common/partition_split.thrift
- src/common/replica_admin.thrift

The file names are descriptive for their content.

backup.thrift / bulkload.thrift / duplication.thrift / partition_split.thrift include the RPCs for the corresponding feature.

consensus.thrift includes the structs for PacificA protocol.

meta_admin.thrift and replica_admin.thrift include the RPCs that can be called by clients to operate on either MetaServer or ReplicaServer for administrative purposes.

metadata.thrift is for structs that can be used for multiple modules. For example, `replica_configuration` are used by `bulkload.thrift` and `consensus.thrift`. They both include "metadata.thrift".

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,6 +41,12 @@ message (STATUS "dsn Installation directory: CMAKE_INSTALL_PREFIX = " ${CMAKE_IN
 # install rdsn/include to ${CMAKE_INCLUDE_PREFIX}/include
 install(DIRECTORY include/ DESTINATION include)
 
+# NOTE: dsn_types.h is not actually a thrift-generated file. As pegasus may reference it,
+# we should copy also this file to the path the same as other thrift-generated files.
```
Comment (Shuo-Jia) [inline CMakeLists.txt:45]: should also copy？

Comment (neverchanje) [reply CMakeLists.txt]: Fixed, any more reviews?

**Review by Shuo-Jia — DISMISSED** [review]

File: src/common/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,63 @@
...
+thrift_generate_cpp(
+    META_ADMIN_THRIFT_SRCS
+    META_ADMIN_THRIFT_HDRS
+    ${CMAKE_CURRENT_SOURCE_DIR}/meta_admin.thrift
+)
+
+thrift_generate_cpp(
+    BULKLOAD_THRIFT_SRCS
+    BULKLOAD_THRIFT_HDRS
+    ${CMAKE_CURRENT_SOURCE_DIR}/bulkload.thrift
```
Comment (hycdong) [inline src/common/CMakeLists.txt:36]: How about using `bulk_load.thrift`?

Comment (neverchanje) [reply src/common/CMakeLists.txt]: Sure. Fixed.

**Review by hycdong — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**